src/dns/LookupDetails.h

```diff
@@ -20,13 +22,21 @@ namespace Dns
 class LookupDetails
 {
 public:
-    LookupDetails() : wait(-1) {} ///< no error, no lookup delay (i.e., no lookup)
-    LookupDetails(const String &anError, int aWait) : error(anError), wait(aWait) {}
+    /// no lookup attempt: no error and no lookup delay
+    LookupDetails(): wait(-1) {}
+
+    /// details a possible lookup attempt
+    /// \param anError either a failed attempt error message or an empty string
+    /// \param aWait \copydoc wait
+    LookupDetails(const SBuf &anError, const int aWait):
+        error(anError.isEmpty() ? std::nullopt : std::make_optional(anError)),
+        wait(aWait)
+    {}
 
     std::ostream &print(std::ostream &os) const;
 
 public:
-    String error; ///< error message for unsuccessful lookups; empty otherwise
+    const std::optional<SBuf> error; ///< error message (if any)
```

Comment:
I added `const` because we do not want some future code to set the `error` member directly (and incorrectly). Folks should go through the constructor.